calico_cni_test.go

```diff
@@ -197,6 +197,59 @@ var _ = Describe("CalicoCni", func() {
 					Expect(err).ShouldNot(HaveOccurred())
 				})
 			})
+
+			Context("when ready flag is false", func() {
+				It("errors when ADD is done", func() {
+					ci, err := calicoClient.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+					r := false
+					ci.Spec.DatastoreReady = &r
+					ci, err = calicoClient.ClusterInformation().Update(ctx, ci, options.SetOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+					_, session, _, _, _, _, err := testutils.CreateContainer(netconf, "", testutils.TEST_DEFAULT_NS, "")
+					Expect(err).Should(HaveOccurred())
+					Eventually(session).Should(gexec.Exit(1))
+				})
+
+				It("errors when DEL is done", func() {
+					_, session, _, _, _, contNs, err := testutils.CreateContainer(netconf, "", testutils.TEST_DEFAULT_NS, "")
+					Expect(err).ShouldNot(HaveOccurred())
+					Eventually(session).Should(gexec.Exit(0))
+
+					ci, err := calicoClient.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+					r := false
+					ci.Spec.DatastoreReady = &r
+					ci, err = calicoClient.ClusterInformation().Update(ctx, ci, options.SetOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+
+					exitCode, err := testutils.DeleteContainer(netconf, contNs.Path(), "", testutils.TEST_DEFAULT_NS)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(exitCode).ShouldNot(Equal(0))
```

Comment:
Nit: I think you forgot to change this line to be should exit 1

Reply:
Since this test is using Expect (not Eventually) I went with what I really wanted to test. It is testing that the result is not success so I just wanted to make sure the exist code is not 0.

Reply:
In other words I don't care about the value except that it is not zero.

Reply:
Oh sorry, didn't see it si not eventually